### ISSUE TITLE
Issue in Differentially expressed gene selection processing

### ISSUE THREAD

**Issue by stevedough123** [dmcable/RCTD#5]
After I'm able to get through the Data Preprocessing steps, I get an error when I try to run the chunk of code in `Differentially expressed gene selection` :

![image](https://user-images.githubusercontent.com/52977783/88250882-d4a79a00-cc5d-11ea-9235-044ca3c7be55.png)

**Comment by lulizou** [comment]
Hi Steve, can you please give more info? i.e. what commands did you run previously, and what is the output of `str(reference)`? Please include the commands/output as code snippets using the "Insert code" button.
Best,
Luli

**Comment by stevedough123** [comment]
Before running this, I ran the data preprocessing steps. However, I got an issue during the `Single-Cell Reference` section:
![image](https://user-images.githubusercontent.com/52977783/88452995-0b291480-ce18-11ea-937b-4240e699b3ac.png)

However, I was still able to save a `SCRef.RDS` file, so I went ahead and ran the next step, but I'm not sure if `SCRef.RDS` was formatted correctly due to the previous error:
![image](https://user-images.githubusercontent.com/52977783/88453005-40356700-ce18-11ea-84fb-18f250a50955.png)

Then, the issue I mentioned in my initial message, occurs when I attempt to run `Differentially expressed gene selection`:
![image](https://user-images.githubusercontent.com/52977783/88453031-62c78000-ce18-11ea-96d5-6a5f331a9c2c.png)

The output of `str(reference)` is:

`Formal class 'Seurat' [package "Seurat"] with 13 slots
  ..@ assays      :List of 1
  .. ..$ RNA:Formal class 'Assay' [package "Seurat"] with 8 slots
  .. .. .. ..@ counts       :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1435200] 23 25 31 71 83 93 142 157 164 179 ...
  .. .. .. .. .. ..@ p       : int [1:1001] 0 1332 2057 2806 5035 6900 7667 9263 10184 10764 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 19327 1000
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:19327] "AL627309" "AL669831" "LINC00115" "FAM41C" ...
  .. .. .. .. .. .. ..$ : chr [1:1000] "BCP6_Total_TCTGAGAAGGTGATTA" "BCP5_Total_CATCAAGTCACCTTAT" "BCP5_Total_CTCCTAGCAGGGTACA" "BCP3_Total_TCGAGGCTCGCCAGCA" ...
  .. .. .. .. .. ..@ x       : num [1:1435200] 1 1 1 6 1 1 1 1 1 1 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ data         :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1435200] 23 25 31 71 83 93 142 157 164 179 ...
  .. .. .. .. .. ..@ p       : int [1:1001] 0 1332 2057 2806 5035 6900 7667 9263 10184 10764 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 19327 1000
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:19327] "AL627309" "AL669831" "LINC00115" "FAM41C" ...
  .. .. .. .. .. .. ..$ : chr [1:1000] "BCP6_Total_TCTGAGAAGGTGATTA" "BCP5_Total_CATCAAGTCACCTTAT" "BCP5_Total_CTCCTAGCAGGGTACA" "BCP3_Total_TCGAGGCTCGCCAGCA" ...
  .. .. .. .. .. ..@ x       : num [1:1435200] 1 1 1 6 1 1 1 1 1 1 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ scale.data   : num[0 , 0 ] 
  .. .. .. ..@ key          : chr "rna_"
  .. .. .. ..@ assay.orig   : NULL
  .. .. .. ..@ var.features : logi(0) 
  .. .. .. ..@ meta.features:'data.frame':	19327 obs. of  0 variables
  .. .. .. ..@ misc         : NULL
  ..@ meta.data   :'data.frame':	1000 obs. of  6 variables:
  .. ..$ orig.ident        : Factor w/ 7 levels "BCP2","BCP3",..: 5 4 4 2 7 4 7 4 4 5 ...
  .. ..$ nCount_RNA        : num [1:1000] 4251 1458 1461 25367 3956 ...
  .. ..$ nFeature_RNA      : int [1:1000] 1332 725 749 2229 1865 767 1596 921 580 933 ...
  .. ..$ cluster           : int [1:1000] 1 2 2 3 4 4 1 4 4 4 ...
  .. ..$ nUMI              : int [1:1000] 4261 1459 1461 25375 3959 1369 4282 2103 1263 2706 ...
  .. ..$ liger_ident_coarse: chr [1:1000] "T cells CD8" "T cells follicular helper" "T cells follicular helper" "Plasma cells" ...
  ..@ active.assay: chr "RNA"
  ..@ active.ident: Factor w/ 7 levels "BCP2","BCP3",..: 5 4 4 2 7 4 7 4 4 5 ...
  .. ..- attr(*, "names")= chr [1:1000] "BCP6_Total_TCTGAGAAGGTGATTA" "BCP5_Total_CATCAAGTCACCTTAT" "BCP5_Total_CTCCTAGCAGGGTACA" "BCP3_Total_TCGAGGCTCGCCAGCA" ...
  ..@ graphs      : list()
  ..@ neighbors   : list()
  ..@ reductions  : list()
  ..@ images      : list()
  ..@ project.name: chr "SeuratProject"
  ..@ misc        : list()
  ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
  .. ..$ : int [1:3] 3 2 0
  ..@ commands    : list()
  ..@ tools       : list()`
